Modules/Events/Signal.lua

```diff
@@ -25,43 +33,37 @@
 			Disconnects all connected events to the signal and voids the signal as unusable.
 --]]
 
-local Signal = {}
-
-function Signal.new()
-	local sig = {}
-	
-	local mSignaler = Instance.new('BindableEvent')
+local function MakeSignal()
 	
-	local mArgData = nil
-	local mArgDataCount = nil
+	local BindableEvent = Instance.new("BindableEvent")
+	local Signal = {}
+	local BindData
 	
-	function sig:fire(...)
-		mArgData = {...}
-		mArgDataCount = select('#', ...)
-		mSignaler:Fire()
+	function Signal:fire(...)
+		BindData = {...}
+		BindableEvent:Fire()
 	end
+	Signal.Fire = Signal.fire
 	
-	function sig:connect(f)
-		if not f then error("connect(nil)", 2) end
-		return mSignaler.Event:connect(function()
-			f(unpack(mArgData, 1, mArgDataCount))
+	function Signal:connect(func)
+		if not func then error("connect(nil)", 2) end
+		return BindableEvent.Event:connect(function()
+			func(unpack(BindData))
 		end)
 	end
 	
-	function sig:wait()
-		mSignaler.Event:wait()
-		assert(mArgData, "Missing arg data, likely due to :TweenSize/Position corrupting threadrefs.")
-		return unpack(mArgData, 1, mArgDataCount)
+	function Signal:wait()
+		BindableEvent.Event:wait()
+		assert(BindData, "Missing arg data, likely due to :TweenSize/Position corrupting threadrefs.")
+		return unpack(BindData)
 	end
 
-	function sig:Destroy()
-		mSignaler:Destroy()
-		mArgData      = nil
-		mArgDataCount = nil
-		mSignaler     = nil
+	function Signal:Destroy()
+		BindData = BindableEvent:Destroy()
```

Comment:
Probably better to just set these values to nil on a separate line to clarify what is happening here. 

Note that :Destroy() is a bit of a nasty method because the connector should be the one disconnecting / handling event connections to avoid keep-alive memory leaks.